daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DFDLFunctions.scala

```diff
@@ -39,6 +41,31 @@ case class DFDLCheckConstraints(recipe: CompiledDPath) extends RecipeOpWithSubRe
   }
 }
 
+case class DFDLCheckRangeInclusive(recipe: CompiledDPath, rangeFrom: Int, rangeTo: Int)
+  extends RecipeOpWithSubRecipes(recipe) {
+
+  override def run(dstate: DState): Unit = {
+    recipe.run(dstate)
+    if (dstate.currentElement.valid.isDefined) {
+      dstate.setCurrentValue(dstate.currentElement.valid.get)
+    } else {
+      val res = executeCheck(dstate, rangeFrom, rangeTo).isOK
+      dstate.currentElement.setValid(true)
+      dstate.setCurrentValue(res)
+    }
+  }
+
+  def executeCheck(dstate: DState, rangeFrom: Integer, rangeTo: Integer): OKOrError = {
+    val testValue: BigInt = dstate.currentValue.getBigInt
+    if (testValue.intValue() >= rangeFrom.intValue() && testValue.intValue() <= rangeTo.intValue()) {
+      OKOrError.OK
+    } else {
+      OKOrError.Error("Error...")
+    }
```

Comment:
This DFDLCheckRange functions don't need to return any error information. Either a value is within range or it's not. It isn't considered an error to not be in range. So this should just return a boolean and setCurrentValue just sets the returned value.

I would also suggest we make some effort to avoid BigInts. Firstly because it's going to be somewhat slow, but secondgly because the currentValue can be one of many different types, including those that aren't integers. This function should be able to statically determine the type and perform an appropriate range change based on that type.